### PR TITLE
Remove state from runtime

### PR DESCRIPTION
**Pull request by bowenwang1996**

### CODE REVIEW

File: node/runtime/src/system.rs [hunk 1]
```diff
@@ -390,6 +392,7 @@ mod tests {
         assert_eq!(apply_results[1].tx_result[0].status, TransactionStatus::Failed);
         assert_eq!(apply_results[2].tx_result[0].status, TransactionStatus::Completed);
         // assert_eq!(apply_result.new_receipts.len(), 0);
+        let viewer = StateDbViewer::new(state_db.clone());
```
Comment (ilblackdragon) [inline node/runtime/src/system.rs]: Out of scope for this, just a question: If runtime doesn't have state should state viewer have it?

Comment (bowenwang1996) [reply node/runtime/src/system.rs]: Yeah I was going to do that in a separate PR. Do you think I should do it in this one?

File: node/runtime/src/lib.rs [hunk 2]
```diff
@@ -856,13 +851,14 @@ impl Runtime {
...
-        &mut self,
+        self,
+        mut state_update: StateDbUpdate,
         apply_state: &ApplyState,
         prev_receipts: &[ReceiptBlock],
         transactions: &[SignedTransaction],
     ) -> ApplyResult {
         let mut new_receipts = HashMap::new();
-        let mut state_update = StateDbUpdate::new(self.state_db.clone(), apply_state.root);
+        // let mut state_update = StateDbUpdate::new(self.state_db.clone(), apply_state.root);
```
[the remark this answers is not in the document]
Comment (bowenwang1996) [reply node/runtime/src/lib.rs]: Removed in b2b5f9f

File: node/runtime/src/lib.rs [hunk 3]
```diff
@@ -945,20 +940,20 @@ impl Runtime {
...
                 &account
             );
         }
-        let (transaction, genesis_root) = state_db_update.finalize();
+        state_update.finalize()
         // TODO: check that genesis_root is not yet in the state_db? Also may be can check before doing this?
-        self.state_db.commit(transaction).expect("Failed to commit genesis state");
-        genesis_root
+        //self.state_db.commit(transaction).expect("Failed to commit genesis state");
+        //genesis_root
```
[the remark this answers is not in the document]
Comment (bowenwang1996) [reply node/runtime/src/lib.rs]: Removed in b2b5f9f

**Pull request merged**